### PR TITLE
Backports for 4-1-stable

### PR DESCRIPTION
**Pull request by maclover7**
Backports commit from Aaron and also #23247.

### CODE REVIEW

File: actionpack/test/controller/render_test.rb [hunk 1]
```diff
@@ -276,7 +281,7 @@ def test_dynamic_render_with_file
   def test_dynamic_render
     assert File.exist?(File.join(File.dirname(__FILE__), '../../test/abstract_unit.rb'))
     assert_raises ActionView::MissingTemplate do
-      get :dynamic_render, { id: '../\\../test/abstract_unit.rb' }
+      get :dynamic_render, params: { id: '../\\../test/abstract_unit.rb' }
```
Comment (tenderlove) [inline actionpack/test/controller/render_test.rb]: `params` doesn't exist on 4-1-stable (I think)

Comment (maclover7) [inline actionpack/test/controller/render_test.rb]: Looks like Action Pack tests are passing on Travis, so I'm going to assume that `params` does exist...?

Comment (kaspth) [inline actionpack/test/controller/render_test.rb]: Please don't. It was added in 5.0 :)

Comment (maclover7) [inline actionpack/test/controller/render_test.rb]: Updated! :)

**Comment by maclover7** [comment]
@kaspth for some reason, removing the `params: {}` is causing the test to fail?

**Comment by kaspth** [comment]
Let's investigate some more then :wink:

**Comment by maclover7** [comment]
Looks like build is passing now

**Comment by rafaelfranca** [comment]
Backported manually in https://github.com/rails/rails/compare/722aa32fa527...00545f3221f7

**Pull request closed without merging**